src/net/connector.rs

```diff
@@ -33,11 +41,20 @@ impl UdpConnector {
 
 impl NetworkConnector for UdpConnector {
     type Stream = UdpSender;
-    
+
     fn connect(&self, host: &str, port: u16, _: &str) -> error::Result<<Self as NetworkConnector>::Stream> {
         let udp_sock = try!(self.0.try_clone());
-        let sock_addr = try!(net::addr_from_trait((host, port)));
-        
+        let sock_addr = match try!(self.local_addr()) {
+            SocketAddr::V4(_) => SocketAddr::V4(SocketAddrV4::new(FromStr::from_str(host).unwrap(), port)),
+            SocketAddr::V6(n) => {
+                let mut addr: SocketAddrV6 = FromStr::from_str(format!("{}:{}", host, port).as_str())
```

Comment:
Fair enough.

Also I found [this](http://servo.github.io/rust-url/url/struct.Url.html#method.host_str) which is most likely where ```hyper``` is getting the incorrect host. However, judging by the comment there, it looks like they intended for the IPv6 host to sit between brackets.